kbs/docs/initdata.md

```diff
@@ -0,0 +1,267 @@
+running in the TEE meets expectations. The softwares to be measured are
+often provided by hardware vendor (like the firmware and tcb security version
+of TEE hardware) or software vendor (like guest kernel for VMs). These
+components are relatively static, which means they may be the same among
+multiple deployments
+
+In some scenarios, users would inject some other information like
+[policy files for kata](https://github.com/kata-containers/kata-containers/blob/main/docs/how-to/how-to-use-the-kata-agent-policy.md),
+[configuration files for components running in guest](https://github.com/confidential-containers/guest-components/tree/main/confidential-data-hub#configuration-file),
+[identity files to specify the identity of TEE](https://github.com/keylime/rust-keylime/blob/master/keylime-agent.conf)
+into the TEE guest when launching.
+
+Compared with static software running in TEE (like guest firmware for TDX VM,
+libos for SGX enclave), these information changes dynamically between different
+deployments and are usually configurations
+We call these information or configurations _Initdata_. Initdata mechanism will
+provides a way to protect their integrity by remote attestation. One thing to note
+is that the confidentiality will not be protected by initdata mechanism because
+the untrusted host can see the plaintext of the data.
+
+To achieve this goal, we defined the following things
+- A data structure named **Initdata Metadata**. This structure is provided by the
+user to contain any initdata in key-value format to untrusted host to inject into
+the TEE when launching. We do not limit the encoding of this data structure, which
+means that JSON and TOML are optional. This will be introduced in [Initdata Metadata](#initdata-metadata)
+- A data integrity binding mechanism. It will guide the untrusted host to bind the
+digest of the `data`` part in the Initdata Metadata to the hardware TEE-specific 
+field in evidence. This field will be checked by the verifier during the remote
+attestation. This will be introduced in 
+[Integrity Binding for Different TEEs](#integrity-binding-for-different-tees)
+- A data serialization method. This method is used to serialize the Initdata Metadata
+into a canonicalized one, which will help to get consistent cryptographic hash for
+different forms of a same Initdata Metadata. This will be introduced in
+[Data Canonicalization Algorithm](#data-canonicalization-algorithm).
+
+This spec does not define how the Initdata Metadata will be delivered into the TEE.
+Different projects will have its own way to do this. For Confidential Containers,
+we will use kata-runtime and kata-agent to collaborate to achieve this function.
+
+## Terminology
+
+This section will introduce the terminology used in this spec to avoid ambiguity.
+
+- `Initdata`: Data that needs to be injected when the TEE is started. This data
+requires integrity protection but does not require confidentiality protection.
```

Comment:
So the InitData isn't the entire structure? It's just the data field?

Reply:
Oh. We had been using the word `initdata` interchangeably in different contexts to refer to both the entire structure and just the data itself before this spec.

My idea is we can accurately define the terms to distinguish `only data field` vs `entire structure := data + metadata`. Maybe we can start with [this thread](https://github.com/confidential-containers/trustee/pull/348#discussion_r1516519010) to have a well-defined terms first. wdyt?

Reply:
I've changed the names.